### ISSUE TITLE
Node "enable_ui_recursively" and "disable_ui_recursively"

### ISSUE THREAD

**Issue by hydroper** [godotengine/godot-proposals#6323]
### Describe the project you are working on

A 2D top-down RPG.

### Describe the problem or limitation you are having in your project

I'm trying to disable or enable control recursively to prevent possible bugs while programmatically animating a panel on its open or collapse transition. That is, for example, I'm trying to disable all buttons under a container, so they can't be pressed.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Provide `enable_ui_recursively` and `disable_ui_recursively` instance methods to `Node`, which are similiar to this:

```gdscript
class_name EnableRecursiveFocus
extends Object

static func enable(node: Node):
	if node is BaseButton:
		node.disabled = false
	elif node is TextEdit:
		node.editable = true
	for c in node.get_children():
		enable(c)

static func disable(node: Node):
	if node is BaseButton:
		node.disabled = true
	elif node is TextEdit:
		node.editable = false
	for c in node.get_children():
		disable(c)
```

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

For example, given you have a container `c`, you can disable it and all of its children recursively with:

```gdscript
c.disable_ui_recursively()
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes. I provided an example implementation above.

### Is there a reason why this should be core and not an add-on in the asset library?

Pratically all Godot projects need this.

**Comment by YuriSizov** [comment]
> Practically all Godot projects need this.

Not really. The code that you've provided, adapted for a particular need of each project, is sufficient enough. You can also use node groups and make calls to all connected nodes with that.

**Comment by KoBeWi** [comment]
You can also do
```GDScript
propagate_call("set_disabled", true)
propagate_call("set_editable", false)
```

**Comment by AThousandShips** [comment]
I believe you need square brackets around the arguments as it takes an array:
```GDScript
propagate_call("set_disabled", [true])
propagate_call("set_editable", [false])
```
